coral/src/app/features/topics/details/components/PromotionBanner.tsx

```diff
@@ -0,0 +1,88 @@
+import { Banner, Box, Button, GridItem } from "@aivenio/aquarium";
+import illustration from "src/app/images/topic-details-schema-Illustration.svg";
+import { ReactElement } from "react";
+import { KlawApiModel } from "types/utils";
+
+interface PromotionBannerProps {
+  // `topicName` is only optional on
+  // KlawApiModel<"PromotionStatus">
+  // so we can't rely on it to be part of
+  // the promotionDetails
+  topicName: string;
+  promotionDetails: KlawApiModel<"PromotionStatus">;
+  type: "schema" | "topic";
+  promoteElement: ReactElement;
+  hasOpenRequest: boolean;
+}
+
+const PromotionBanner = ({
+  promotionDetails,
+  hasOpenRequest,
+  type,
+  promoteElement,
+  topicName,
```

Comment:
I think this should probably be `entityName`, as connectors can also be promoted, and we probably want to use this banner for them as well :o

<img width="1318" alt="Screenshot 2023-07-26 at 14 43 40" src="https://github.com/aiven/klaw/assets/20607294/9655537d-d3d3-4171-904b-da02e9b10074">

Reply:
That's a good idea! And if the optimisation is renaming a property, I think it's ok even if it may be premature ^^

Reply:
https://github.com/aiven/klaw/pull/1513/commits/8ab3773e7be40ee3dd36629d0b0457b399a43081